### DISCUSSION TITLE
Persistent .on Method Error with Supabase Realtime Subscriptions in React

### DISCUSSION THREAD

**Discussion by kh1z3r** [supabase/supabase#23252] [Questions] [answered]
Hello,

I'm encountering a problem when setting up a real-time subscription to listen for `INSERT` events on a comments table using Supabase in a React component. I have set up my subscription within a `useEffect` hook, but I'm running into issues with receiving updates. The `on` method doesn't seem to trigger or is not recognized as a function. Below is the snippet of my `Comments` component where the subscription is implemented

The error I'm receiving is TypeError: (intermediate value).from(...).on is not a function, which occurs on the line where the on method is called. This is perplexing as the same pattern is documented in Supabase examples.

I'm using the following versions:

    @supabase/supabase-js: version 2.42.5
    react: version 18.2.0

Has anyone faced a similar issue or can spot what might be going wrong here? Any help would be greatly appreciated!

Thank you!

```jsx
// Comments.jsx
import React, { useState, useEffect } from 'react';
import { supabase } from '../utils/supabaseClient';
import useUserId from '../utils/useUserId';
import '../styles/Comments.css';

function Comments({ postId }) {
  // ... state initialization ...

  useEffect(() => {
    fetchComments();

    const commentsSubscription = supabase
      .from(`comments:post_id=eq.${postId}`)
      .on('*', payload => {
        console.log('Change received:', payload.new);
        if (payload.eventType === 'INSERT' && payload.new.post_id === postId) {
          setComments(prevComments => [...prevComments, payload.new]);
        }
      })
      .subscribe();

    // Cleanup function
    return () => {
      supabase.removeSubscription(commentsSubscription);
    };
  }, [postId]);

  // ... rest of the component ...
}

export default Comments;
'''

**Comment by GaryAustin1** [comment, marked as answer]
As mentioned in Discord that is not the correct realtime syntax for supabase-js 2.x.x.  It is a very old supabase-js version 1 method.

**Reply by GaryAustin1** [reply]
![image](https://github.com/supabase/supabase/assets/54564956/95a0dfae-85f9-449d-8e81-6fb053b18df8)
Is the current syntax.
